### ISSUE TITLE
Pigeons next_exec_folder sometimes initializes seed with negative numbers

### ISSUE THREAD

**Issue by dchang10** [Julia-Tempering/Pigeons.jl#150]
Running Pigeonsv0.2.6 with checkpointing enabled sometimes causes an error where the random number generator is seeded with a negative number during the `next_exec_folder` call.

https://github.com/Julia-Tempering/Pigeons.jl/blob/295f36848b673156dc3c071a3945ce60a3be685a/src/utils/exec_folder.jl#L10

```
result = pigeons(
                  target = toy_mvn_target(100),
                  checkpoint = true)
ERROR: DomainError with -3773815612603080421:
'n' must be non-negative.
Stacktrace:
  [1] make_seed(n::Int64)
    @ Random ~/.julia/juliaup/julia-1.9.3+0.x64.linux.gnu/share/julia/stdlib/v1.9/Random/src/RNGs.jl:294
  [2] seed!
    @ ~/.julia/juliaup/julia-1.9.3+0.x64.linux.gnu/share/julia/stdlib/v1.9/Random/src/Xoshiro.jl:156 [inlined]
  [3] seed!
    @ ~/.julia/juliaup/julia-1.9.3+0.x64.linux.gnu/share/julia/stdlib/v1.9/Random/src/RNGs.jl:358 [inlined]
  [4] seed!
    @ ~/.julia/juliaup/julia-1.9.3+0.x64.linux.gnu/share/julia/stdlib/v1.9/Random/src/RNGs.jl:363 [inlined]
  [5] next_exec_folder()
    @ Pigeons ~/.julia/packages/Pigeons/RtdCL/src/utils/exec_folder.jl:10
  [6] pt_exec_folder
    @ ~/.julia/packages/Pigeons/RtdCL/src/pt/PT.jl:56 [inlined]
  [7] PT(inputs::Inputs{Pigeons.ScaledPrecisionNormalPath, Nothing, Nothing, Nothing}; exec_folder::String)
    @ Pigeons ~/.julia/packages/Pigeons/RtdCL/src/pt/PT.jl:49
  [8] PT
    @ ~/.julia/packages/Pigeons/RtdCL/src/pt/PT.jl:46 [inlined]
  [9] pigeons(pt_arguments::Inputs{Pigeons.ScaledPrecisionNormalPath, Nothing, Nothing, Nothing}, #unused#::Pigeons.ThisProcess)
    @ Pigeons ~/.julia/packages/Pigeons/RtdCL/src/api.jl:19
 [10] #pigeons#155
    @ ~/.julia/packages/Pigeons/RtdCL/src/api.jl:16 [inlined]
 [11] top-level scope
    @ REPL[7]:1
```

For example `getpid() * now().instant.periods.value` in my current instance returns -3773814961396238284 .

**Comment by alexandrebouchard** [comment]
Interesting! Random.seed! does not document it has to be given a positive argument (and linux pids are suppose to be positive! probably an integer underflow)

Nonetheless I have pushed a PR with an easy fix, relying on our own rng instead of Julia's. Should be available soon.

**Comment by alexandrebouchard** [comment]
OK fixed in the head of main! Version 0.2.7 to follow later today... Thanks again!
